src/br/unicamp/meca/mind/MecaMind.java

```diff
@@ -277,140 +293,189 @@ private void mountMotivationalCodelets() {
-									reactiveBehavioralCodelet.addOutputs(motorCodelet.getInputs());
+										.equalsIgnoreCase(actionCodelet.getMotorCodeletId())) {
+									actionCodelet.addOutputs(motorCodelet.getInputs());
 								}
 							}
 						}
 					}
 					/*
 					 * Inputs
 					 */
+					
+					if (motivationalCodelets != null) {
+						for (MotivationalCodelet motivationalCodelet : motivationalCodelets) {
+							if (motivationalCodelet != null && motivationalCodelet.getId() != null) {
+								ArrayList<String> motivationalCodeletsIds = actionCodelet
+										.getMotivationalCodeletsIds();
+								if (motivationalCodeletsIds != null) {
+									for (String motivationalCodeletId : motivationalCodeletsIds) {
+										if (motivationalCodeletId != null && motivationalCodelet.getId()
+												.equalsIgnoreCase(motivationalCodeletId)) {
+											actionCodelet.addInputs(motivationalCodelet.getOutputs());
+										}
+									}
+								}
+							}
+						}
+					}
+					
 					if (perceptualCodelets != null) {
 						for (PerceptualCodelet perceptualCodelet : perceptualCodelets) {
 							if (perceptualCodelet != null && perceptualCodelet.getId() != null) {
-								ArrayList<String> perceptualCodeletsIds = reactiveBehavioralCodelet
+								ArrayList<String> perceptualCodeletsIds = actionCodelet
 										.getPerceptualCodeletsIds();
 								if (perceptualCodeletsIds != null) {
 									for (String perceptualCodeletId : perceptualCodeletsIds) {
 										if (perceptualCodeletId != null
 												&& perceptualCodelet.getId().equalsIgnoreCase(perceptualCodeletId)) {
-											reactiveBehavioralCodelet.addInputs(perceptualCodelet.getOutputs());
+											actionCodelet.addInputs(perceptualCodelet.getOutputs());
 										}
 									}
 								}
 							}
 						}
 					}
 
-					if (soarCodelet != null && soarCodelet.getId() != null && reactiveBehavioralCodelet.getSoarCodeletId() != null) {
-						if (soarCodelet.getId().equalsIgnoreCase(reactiveBehavioralCodelet.getSoarCodeletId())) {
-							reactiveBehavioralCodelet.addBroadcasts(soarCodelet.getOutputs());
+					if (soarCodelet != null && soarCodelet.getId() != null && actionCodelet.getSoarCodeletId() != null) {
+						if (soarCodelet.getId().equalsIgnoreCase(actionCodelet.getSoarCodeletId())) {
+							actionCodelet.addBroadcasts(soarCodelet.getOutputs());
 						}
 
-					}
-
+					}				
 				}
 			}
 		}
+		
 	}
 
-	private void mountRandomBehavioralCodelets() {
-		if (getRandomBehavioralCodelets() != null) {
-			for (RandomBehavioralCodelet randomBehavioralCodelet : getRandomBehavioralCodelets()) {
-				if (randomBehavioralCodelet != null && randomBehavioralCodelet.getId() != null && randomBehavioralCodelet.getMotorCodeletId() != null) {
-					insertCodelet(randomBehavioralCodelet);
+	private void mountActionFromPlanningCodelets() {
+		if (actionFromPlanningCodelets != null) {
+			for (ActionFromPlanningCodelet actionCodelet : actionFromPlanningCodelets) {
```

Comment:
actionFromPlanningCodelets.stream().forEach(........

Reply:
Nice suggestion, indeed, but as is just a matter of style, resulting in the same end, will let it to a future PR.